### DISCUSSION TITLE
Disable file watch capabilities

### DISCUSSION THREAD

**Discussion by rymndhng** [joaotavora/eglot#1128] [Q&A]
Is there some way to disable the file watching capabilities?

For some context, I'm working in a large codebase on a mac (using pyright lsp server) and the capability `workspace/didChangeWatchedFiles` is crashing the subprocess b/c I've running out of file descriptors -- I believe this is due to the use of `kqueue` which consumes one fd per file to watch..

So far I've added this into my init.el script ... is there an official way of turning this off?
```
  (cl-defmethod eglot-register-capability
    (server (method (eql workspace/didChangeWatchedFiles)) id &key watchers)
    nil)
```

**Comment by joaotavora** [comment]
This is a good way to do these exceptions.  You can also peek into
'server''s
properties if you want to turn this off only for a specific context.  Or
you
could subclass eglot-lsp-server and specialize the function just for that.
Or you could ask the specific server not enable that capability request.

Finally, Eglot could check eglot-ignored-server-capabilities before
allowing that capability to be registered. But this would affect all servers
and all projects.  Nevertheless, if you think this is a more correct way,
please submit a bug report to ***@***.*** and CC me
in the email.
